library/blob.c

```diff
@@ -491,6 +550,22 @@ static int eblob_check_disk_one(struct eblob_iterate_local *loc)
 		}
 	}
 
+	if ((ctl->flags & EBLOB_ITERATE_FLAGS_VERIFY_CHECKSUM) &&
+	    !(dc->flags & BLOB_DISK_CTL_REMOVE) &&
+	    !(dc->flags & BLOB_DISK_CTL_UNCOMMITTED)) {
+		struct eblob_write_control wc;
+		memset(&wc, 0, sizeof(wc));
+		eblob_rctl_to_wc(&rc, &wc);
+		eblob_dc_to_wc(dc, &wc);
+
+		err = eblob_verify_checksum(bc->back, &dc->key, &wc);
+		if (err) {
+			eblob_dump_wc(bc->back, &dc->key, &wc, "eblob_check_disk_one: checksum verification failed", err);
+			err = 0;
+			goto err_out_exit;
+		}
+	}
+
```

Comment:
That clarifies things!
Please add comments about this issue (returned error, call path unwind and its possible consequences) so that it would not rise questions next time